web/src/components/zonemap.jsx

```diff
@@ -107,22 +106,62 @@ const ZoneMap = ({
       hover: { 'fill-color': 'white', 'fill-opacity': 0.3 },
       ocean: { 'background-color': theme.oceanColor },
       zonesBorder: { 'line-color': theme.strokeColor, 'line-width': theme.strokeWidth },
-      zonesClickable: { 'fill-color': ['case', ['has', 'color'], ['get', 'color'], theme.clickableFill] },
+      zonesClickable: {
+        'fill-color': [
+          'coalesce', // // https://docs.mapbox.com/mapbox-gl-js/style-spec/expressions/#coalesce
+          ['feature-state', 'color'],
+          ['get', 'color'],
+          theme.clickableFill,
+        ],
+      },
       zonesNonClickable: { 'fill-color': theme.nonClickableFill },
     }),
-    [theme],
+    [theme]
   );
 
   // If WebGL is not supported trigger an error callback.
-  useEffect(
-    () => {
-      if (!ReactMapGL.supported()) {
-        setIsSupported(false);
-        onMapError('WebGL not supported');
-      }
-    },
-    [],
-  );
+  useEffect(() => {
+    if (!ReactMapGL.supported()) {
+      setIsSupported(false);
+      onMapError('WebGL not supported');
+    }
+  }, []);
+
+  // TODO: Consider moving the calculation to a useMemo function
+  // change color of zones if timeslider is changed
+  useEffect(() => {
+    if (isHistoryFeatureEnabled && isLoaded && co2ColorScale) {
+      // TODO: This will only change RENDERED zones, so if you change the time in Europe and zoom out, go to US, it will not be updated!
+      // TODO: Consider using isdragging or similar to update this when new zones are rendered
+      const features = ref.current.queryRenderedFeatures();
```

Comment:
Perhaps we should just initially start by querying all features here. I did so in my prototype with quite good performance.

Then we can optimize it through dragging if we'd like later on, I'm worried this will cause some odd experiences.

Reply:
Hmm, in your prototype (on `mk/map-color-feature-states`) we also change only color of the zones that are rendered as far as I can tell :)

https://user-images.githubusercontent.com/3296643/166935649-98080222-b448-4b13-bc81-5898c32385a6.mp4

Reply:
Thanks for testing and good point, that definitely solves some of the problems - let's roll with it! :)